Add sensor dts support for nxp boards

MaureenHelm opened the pull request:
- Adds dts yaml bindings for the fxos8700, fxas21002, and max30101 sensors
- Adds dts nodes and fixups to frdm_k64f, frdm_kw41z, and hexiwear_k64 boards
- ~~Fixes some board dependencies in sanitycheck, related to converting hts221 and lis3mdl sensors to dts~~

MaureenHelm committed:
frdm_k64f: Add sensor dts nodes and fixups

Adds dts nodes and fixups for the fxos8700 sensor to the frdm_k64f
board. Tested with samples/sensor/fxos8700

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
frdm_kw41z: Add sensor dts nodes and fixups

Adds dts nodes and fixups for the fxos8700 sensor to the frdm_kw41z
board. Tested with samples/sensor/fxos8700

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
hexiwear_k64: Add sensor dts nodes and fixups

Adds dts nodes and fixups for the fxos8700, fxas21002, and max30101
sensors to the hexiwear_k64 board. Tested with:
- samples/sensor/fxos8700
- samples/sensor/fxas21002
- samples/sensor/max30101

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
dts: sensor: Add dts yaml bindings for the fxos8700 sensor

Adds dts yaml bindings for the fxos8700 sensor to allow the sensor
driver name, i2c driver name, and i2c address to be extracted from dts
instead of kconfig.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
dts: sensor: Add dts yaml bindings for the fxas21002 sensor

Adds dts yaml bindings for the fxas21002 sensor to allow the sensor
driver name, i2c driver name, and i2c address to be extracted from dts
instead of kconfig.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
dts: sensor: Add dts yaml bindings for the max30101 sensor

Adds dts yaml bindings for the max30101 sensor to allow the sensor
driver name, i2c driver name, and i2c address to be extracted from dts
instead of kconfig.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm committed:
samples: Build hts221 sensor sample only on supported boards

The hts221 sensor driver was implicitly made dependent upon the
disco_l475_iot1 board when it was converted to dts, so build the
associated sample only for this board. This issue was encountered when
adding sensor dts support to nxp boards, which don't have hts221.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

erwango commented on tests/drivers/build_all/sensors_a_m.conf:
```diff
@@ -18,11 +18,11 @@ CONFIG_DHT=y
 CONFIG_HMC5883L=y
-CONFIG_HTS221=y
+# CONFIG_HTS221=y
 CONFIG_HP206C=y
```
Can you check this is not the case anymore on recent master branch?
I've reverted some Kconfig change for these sensor to remove the dependency on disco board.

MaureenHelm replied:
Do you mean 2867b86 and 11eac23? Yes, these are included. The issue is that when I select `HAS_DTS_I2C_DEVICE` on a board that does not have hts221, the compile fails because `CONFIG_HTS221_NAME` and `CONFIG_HTS221_I2C_MASTER_DEV_NAME` are not defined. Same is true for lis3mdl.

erwango replied:
I see. But then once we end up with all sensors converted to DTS, we'll have the issue with all the sensors. If possible, I think this dependency should be fixed rather than by-passed. I'll have a look

erwango replied:
OK, I see now.
So this will be the case for all sensors converted to DTS not present on a board enabling HAS_DTS_I2C_DEVICE.
This is because HAS_DTS_I2C_DEVICE will prevent from inheriting of sensor's Kconfig default values.
So this is a general incompatibility of dts and "build all on all boards" kind of tests. Since dts will be more and more present, this is worse being addressed or at least discussed in a dedicated issue.

erwango commented on boards/arm/disco_l475_iot1/disco_l475_iot1.yaml:
```diff
@@ -5,3 +5,6 @@ arch: arm
 toolchain:
   - zephyr
   - gccarmemb
+supported:
```
cf comment on previous commit.
Commit might not be needed

erwango reviewed (dismissed):
Update on HTS221 and LIS3MDL recent changes to simplify the PR
Otherwise LGTM

galak commented on tests/drivers/build_all/sensors_a_m.conf:
```diff
@@ -18,11 +18,11 @@ CONFIG_DHT=y
 CONFIG_HDC1008=y
 CONFIG_HMC5883L=y
-CONFIG_HTS221=y
+# CONFIG_HTS221=y
```
With my recent changes we should be able to drop this commit once they go in.

galak commented on drivers/sensor/fxos8700/Kconfig:
```diff
@@ -12,6 +12,8 @@ menuconfig FXOS8700
 	  Enable driver for the FXOS8700 accelerometer/magnetometer
 
+if !HAS_DTS_I2C_DEVICE
+
```
I can't comment on the proper line, but can you remove the !HAS_DTS_I2C, now that this is wrapped in a !HAS_DTS_I2C_DEVICE

galak commented on drivers/sensor/max30101/Kconfig:
```diff
@@ -13,6 +13,8 @@ menuconfig MAX30101
 if MAX30101
 
+if !HAS_DTS_I2C_DEVICE
+
```
I can't comment on the proper line, but can you remove the !HAS_DTS_I2C, now that this is wrapped in a !HAS_DTS_I2C_DEVICE

galak commented on drivers/sensor/fxas21002/Kconfig:
```diff
@@ -14,6 +14,8 @@ menuconfig FXAS21002
 if FXAS21002
 
+if !HAS_DTS_I2C_DEVICE
+
```
I can't comment on the proper line, but can you remove the !HAS_DTS_I2C, now that this is wrapped in a !HAS_DTS_I2C_DEVICE

galak reviewed (changes requested):
We can remove !HAS_DTS_I2C for things wrapped with !HAS_DTS_I2C_DEVICE

MaureenHelm committed:
tests: build_all: Add missing sensors

Now that we have an application level dts fixup, we can re-enable the
fxos8700 sensor in the build_all test. Also added fxas21002 and max30101
sensors which were missing.

Signed-off-by: Maureen Helm <maureen.helm@nxp.com>

MaureenHelm commented:
- Rebased to latest master
- Removed HTS221 and LIS3MDL workarounds
- Updated build_all test with missing sensors

MaureenHelm commented:
samples/sensor/hts221 is failing again. My original solution was to add `depends on: hts221` to the sample.yaml, but this could also be solved by adding an application level dts fixup. @galak, which do you prefer?

galak commented:
I think a depends on: hts221 makes sense, since the sample/test doesn't make any sense unless you have the sensor.

MaureenHelm commented:
> I think a depends on: hts221 makes sense, since the sample/test doesn't make any sense unless you have the sensor.

I was leaning that way too. Cherry-picked that commit (7444944) back in.
